File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        if (root == null) return [];
        let queue = [root];
        let r = [];
        while (queue.length != 0){
            let arr = [];
            let level = queue.length;
            for (let i = 0; i < level; i++){
                let first = queue.shift();
                arr.push(first.val);
                if (first.left) {queue.push(first.left);}
                if (first.right) {queue.push(first.right);}
            }
            r.push(arr);
        }
        return r;
    }
}

File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        m = [-1001]
        def dfs(root):
            if root is None:
                return 0
            left = dfs(root.left)
            right = dfs(root.right)
            m[0] = max(m[0], root.val+left+right, root.val)
            return root.val + max(left, right)
        dfs(root)
        return m[0]